pkg/compactor/executor_test.go

```diff
@@ -0,0 +1,438 @@
+		MaxClosingBlocksConcurrency:         1,
+		SymbolsFlushersConcurrency:          1,
+		MaxBlockUploadValidationConcurrency: 1,
+		BlockRanges:                         mimir_tsdb.DurationList{2 * time.Hour, 12 * time.Hour, 24 * time.Hour},
+	}
+}
+
+// mockCompactorSchedulerClient implements CompactorSchedulerClient
+type mockCompactorSchedulerClient struct {
+	mu                 sync.Mutex
+	leaseJobCallCount  int
+	updateJobCallCount int
+	firstUpdate        schedulerpb.UpdateType
+	lastUpdate         schedulerpb.UpdateType
+	recvPlannedReq     bool
+	LeaseJobFunc       func(ctx context.Context, in *schedulerpb.LeaseJobRequest) (*schedulerpb.LeaseJobResponse, error)
+	UpdateJobFunc      func(ctx context.Context, in *schedulerpb.UpdateCompactionJobRequest) (*schedulerpb.UpdateJobResponse, error)
+	PlannedJobsFunc    func(ctx context.Context, in *schedulerpb.PlannedJobsRequest) (*schedulerpb.PlannedJobsResponse, error)
+	UpdatePlanJobFunc  func(ctx context.Context, in *schedulerpb.UpdatePlanJobRequest) (*schedulerpb.UpdateJobResponse, error)
+}
+
+func (m *mockCompactorSchedulerClient) LeaseJob(ctx context.Context, in *schedulerpb.LeaseJobRequest, opts ...grpc.CallOption) (*schedulerpb.LeaseJobResponse, error) {
+	m.mu.Lock()
+	m.leaseJobCallCount++
+	m.mu.Unlock()
+	return m.LeaseJobFunc(ctx, in)
+}
+
+func (m *mockCompactorSchedulerClient) UpdateCompactionJob(ctx context.Context, in *schedulerpb.UpdateCompactionJobRequest, opts ...grpc.CallOption) (*schedulerpb.UpdateJobResponse, error) {
+	m.mu.Lock()
+	if m.updateJobCallCount == 0 {
+		m.firstUpdate = in.Update
+	}
+	m.lastUpdate = in.Update
+	m.updateJobCallCount++
+	m.mu.Unlock()
+	return m.UpdateJobFunc(ctx, in)
+}
+
+func (m *mockCompactorSchedulerClient) PlannedJobs(ctx context.Context, in *schedulerpb.PlannedJobsRequest, opts ...grpc.CallOption) (*schedulerpb.PlannedJobsResponse, error) {
+	m.mu.Lock()
+	m.recvPlannedReq = true
+	m.mu.Unlock()
+	return m.PlannedJobsFunc(ctx, in)
+}
+
+func (m *mockCompactorSchedulerClient) UpdatePlanJob(ctx context.Context, in *schedulerpb.UpdatePlanJobRequest, opts ...grpc.CallOption) (*schedulerpb.UpdateJobResponse, error) {
+	m.mu.Lock()
+	if m.updateJobCallCount == 0 {
+		m.firstUpdate = in.Update
+	}
+	m.lastUpdate = in.Update
+	m.updateJobCallCount++
+	m.mu.Unlock()
+	return m.UpdatePlanJobFunc(ctx, in)
+}
+
+func (m *mockCompactorSchedulerClient) GetFirstUpdate() schedulerpb.UpdateType {
+	m.mu.Lock()
+	defer m.mu.Unlock()
+	return m.firstUpdate
+}
+
+func (m *mockCompactorSchedulerClient) GetLastUpdate() schedulerpb.UpdateType {
+	m.mu.Lock()
+	defer m.mu.Unlock()
+	return m.lastUpdate
+}
+
+func (m *mockCompactorSchedulerClient) ReceivedPlannedRequest() bool {
+	m.mu.Lock()
+	defer m.mu.Unlock()
+	return m.recvPlannedReq
+}
+
+func (m *mockCompactorSchedulerClient) GetUpdateJobCallCount() int {
+	m.mu.Lock()
+	defer m.mu.Unlock()
+	return m.updateJobCallCount
+}
+
+func TestSchedulerExecutor_JobStatusUpdates(t *testing.T) {
+
```

Comment:
Total nit - but I notice a blank line at the beginning of functions here and there in this PR. Is this intended?